MODULE.bazel

```diff
@@ -0,0 +1,20 @@
+module(
+    name = "rules_m4",
+    version = "0.2.2-bcr.0",
+    compatibility_level = 1,
+)
+
+bazel_dep(
+    name = "googletest",
+    version = "1.11.0",
+    repo_name = "com_google_googletest",
+    dev_dependency = True,
+)
+
+initialize_m4_toolchains = use_extension(
+    "@rules_m4//m4:extensions.bzl",
+    "initialize_m4_toolchains",
+)
+initialize_m4_toolchains.configure(extra_copts = [])
+use_repo(initialize_m4_toolchains, "m4_v1.4.18")
+register_toolchains("@rules_m4//m4/toolchains:v1.4.18")
```

Comment:
I'm not familiar enough with the bzlmod "extensions" API to say for sure, but would the following code also work?
```python
m4 = use_extension("@rules_m4//m4:extensions.bzl", "m4_extension")
m4.register_toolchains()  # using default options
m4.register_toolchains(version = "1.4.17", extra_copts = ["-w"])
```

Where the extension looks something like this:
```python
def _m4_extension(module_ctx):
    for module in module_ctx.modules:
        for register in module.tags.register_toolchains:
            check_version(register.version)
            repo_name = "m4_v{}".format(register.version)
            m4_repository(
                name = repo_name,
                version = register.version,
                extra_copts = register.extra_copts,
            )

            native.register_toolchains("@rules_m4//m4/toolchains:v{}".format(version))
```

Reply:
I'll get back to you on this one. I'll check tomorrow.